src/deepsparse/loggers/helpers.py

```diff
@@ -285,13 +280,39 @@ def check_identifier_match(
     if template[:3] == "re:":
         pattern = template[3:]
         return re.match(pattern, identifier) is not None, None
-    if template == identifier:
-        return True, None
-    if template.startswith(identifier):
-        remainder = template.replace(identifier, "")
-        return True, remainder if remainder.startswith("[") else remainder[1:]
-    if template in identifier:
-        return True, None
+
+    match = SequenceMatcher(None, identifier, template).find_longest_match(
+        0, len(identifier), 0, len(template)
+    )
+    if not match:
+        return False, None
+
+    if match.b == 0:
```

Comment:
Needed to refactor here to account for the fact, that when we create a logger the pipeline is not yet existing. This means that we need to able to match identifiers against templates that do not have pipeline name e.g:
template: `something.sequence[0]`
identifier: `pipeline_name/something`
should result in match and reminder `sequence[0]`

Reply:
Why not keep this solution? It is not cluttering the code or its readability. On the other hand, the solutions that you are proposing are quite elegant, but make the process more complicated for the user.